src/components/input/_input.scss

```
@@ -86,6 +86,16 @@
     margin-top: 0.25rem;
 }
 
+.ons-input__header-search {
```

```suggestion
.ons-input--header-search {
```
I think this should be named like this to follow BEM